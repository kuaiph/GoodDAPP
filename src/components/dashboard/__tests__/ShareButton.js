import React from 'react'
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'
import mdd from 'mobile-device-detect'

import { getWebRouterComponentWithMocks } from './__util__'

describe('ShareButton', () => {
  afterEach(() => {
    mdd.isMobile = false
  })

  it('renders without errors for a mobile device', () => {
    // Given
    mdd.isMobile = true
    const ShareButton = getWebRouterComponentWithMocks('../ShareButton')
    const component = renderer.create(<ShareButton />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toBeTruthy()
  })

  it('renders null for a non mobile device', () => {
    // Given
    const ShareButton = getWebRouterComponentWithMocks('../ShareButton')
    const component = renderer.create(<ShareButton />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toBeNull()
  })

  it('matches snapshot for a link without destination', () => {
    // Given
    mdd.isMobile = true
    const ShareButton = getWebRouterComponentWithMocks('../ShareButton')
    const props = { screenProps: { screenState: { sendLink: 'https://example.com/shareMyLink', to: '' } } }
    const component = renderer.create(<ShareButton {...props} />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot for a link with destination', () => {
    // Given
    mdd.isMobile = true
    const ShareButton = getWebRouterComponentWithMocks('../ShareButton')
    const props = { screenProps: { screenState: { sendLink: 'https://example.com/shareMyLink', to: 'me@example.co' } } }
    const component = renderer.create(<ShareButton {...props} />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
