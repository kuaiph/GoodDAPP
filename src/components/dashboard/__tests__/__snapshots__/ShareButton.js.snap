// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareButton matches snapshot for a link with destination 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxPack": "center",
      "WebkitJustifyContent": "center",
      "backgroundColor": "rgba(85,85,85,1.00)",
      "borderBottomColor": "rgba(0,0,0,0.00)",
      "borderBottomLeftRadius": "4px",
      "borderBottomRightRadius": "4px",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftColor": "rgba(0,0,0,0.00)",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightColor": "rgba(0,0,0,0.00)",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopColor": "rgba(0,0,0,0.00)",
      "borderTopLeftRadius": "4px",
      "borderTopRightRadius": "4px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
      "justifyContent": "center",
      "marginTop": "1em",
      "minWidth": "64px",
      "msFlexPack": "center",
    }
  }
>
  <div
    className="css-cursor-18t94o4 css-view-1dbjc4n"
    data-focusable={true}
    disabled={false}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    style={
      Object {
        "borderBottomLeftRadius": "4px",
        "borderBottomRightRadius": "4px",
        "borderTopLeftRadius": "4px",
        "borderTopRightRadius": "4px",
        "cursor": "pointer",
        "msTouchAction": "manipulation",
        "overflowX": "hidden",
        "overflowY": "hidden",
        "position": "relative",
        "touchAction": "manipulation",
      }
    }
    tabIndex="0"
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitBoxPack": "center",
          "WebkitFlexDirection": "row",
          "WebkitJustifyContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "msFlexAlign": "center",
          "msFlexDirection": "row",
          "msFlexPack": "center",
        }
      }
    >
      <div
        className="css-text-76zvg2 css-textOneLine-bfa6kz"
        dir="auto"
        style={
          Object {
            "color": "rgba(255,255,255,1.00)",
            "direction": "ltr",
            "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "letterSpacing": "1px",
            "marginBottom": "9px",
            "marginLeft": "16px",
            "marginRight": "16px",
            "marginTop": "9px",
            "textAlign": "center",
          }
        }
      >
        <span
          className="css-text-76zvg2 css-textHasAncestor-16my406"
          dir="auto"
          style={
            Object {
              "direction": "ltr",
              "fontFamily": "Helvetica, \\"sans-serif\\"",
              "fontSize": "18px",
              "fontWeight": "bold",
              "paddingBottom": "5px",
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "paddingTop": "5px",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        >
          Share Link
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`ShareButton matches snapshot for a link without destination 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxPack": "center",
      "WebkitJustifyContent": "center",
      "backgroundColor": "rgba(85,85,85,1.00)",
      "borderBottomColor": "rgba(0,0,0,0.00)",
      "borderBottomLeftRadius": "4px",
      "borderBottomRightRadius": "4px",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftColor": "rgba(0,0,0,0.00)",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightColor": "rgba(0,0,0,0.00)",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopColor": "rgba(0,0,0,0.00)",
      "borderTopLeftRadius": "4px",
      "borderTopRightRadius": "4px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
      "justifyContent": "center",
      "marginTop": "1em",
      "minWidth": "64px",
      "msFlexPack": "center",
    }
  }
>
  <div
    className="css-cursor-18t94o4 css-view-1dbjc4n"
    data-focusable={true}
    disabled={false}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    style={
      Object {
        "borderBottomLeftRadius": "4px",
        "borderBottomRightRadius": "4px",
        "borderTopLeftRadius": "4px",
        "borderTopRightRadius": "4px",
        "cursor": "pointer",
        "msTouchAction": "manipulation",
        "overflowX": "hidden",
        "overflowY": "hidden",
        "position": "relative",
        "touchAction": "manipulation",
      }
    }
    tabIndex="0"
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitBoxPack": "center",
          "WebkitFlexDirection": "row",
          "WebkitJustifyContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "msFlexAlign": "center",
          "msFlexDirection": "row",
          "msFlexPack": "center",
        }
      }
    >
      <div
        className="css-text-76zvg2 css-textOneLine-bfa6kz"
        dir="auto"
        style={
          Object {
            "color": "rgba(255,255,255,1.00)",
            "direction": "ltr",
            "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "letterSpacing": "1px",
            "marginBottom": "9px",
            "marginLeft": "16px",
            "marginRight": "16px",
            "marginTop": "9px",
            "textAlign": "center",
          }
        }
      >
        <span
          className="css-text-76zvg2 css-textHasAncestor-16my406"
          dir="auto"
          style={
            Object {
              "direction": "ltr",
              "fontFamily": "Helvetica, \\"sans-serif\\"",
              "fontSize": "18px",
              "fontWeight": "bold",
              "paddingBottom": "5px",
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "paddingTop": "5px",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        >
          Share Link
        </span>
      </div>
    </div>
  </div>
</div>
`;
